### ISSUE TITLE
[Audit_RV] Suggested code improvements

### ISSUE THREAD

**Issue by yanliu18** [galacticcouncil/HydraDX-math#51]
During the audit, we discovered a few improvements that could improve the efficiency of computation or have better precision. Here is a list:

- [x] Duplicated computation of `protocol_fee_amount`.
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L53
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L38

The variable `protocol_fee_amount` is calculated in line 38 and 53 repeatedly. Only one is needed and the result could be reused.

- [x] repeated computation of variables regarding price difference of an asset at the time of invest-in and cash-out.
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L289
let `alpha` = `sub.checked_div(&sum)`
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L358
We know `alpha + p1 = 1`, such that one could be calculated from the other. Since unsigned data is used throughout the implementation, it is recommended to compute `p1` (could have a self-explaining variable name) at the beginning. And use `1-p1` to compute `alpha` when needed.

- [x] Repeated computation of `delta_reserve` in the following code
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L364
https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L338
`p2` = `delta_reserve` since when price raised, `delta_b = 0`

- [x] combine the above simplification (2-3), starting line 354 to 368 can be safely reduced to one line as follows:
 https://github.com/galacticcouncil/HydraDX-math/blob/f4dab244bb5d57971325f0ba6fbb8f6c1353beeb/src/omnipool/math.rs#L354

```rust
current_price.checked_mul_int(FixedU128::One.checked_sub(p1).checked_mul(delta_reserve)?)?
```
Provided `p1` is computed as the very beginning of `calculate_remove_liquidity_state_changes`

**Comment by enthusiastmartin** [comment]
First suggestion is refactored in #54 

The second and third - these parts have been refactored in #53

**Comment by enthusiastmartin** [comment]
fixed in #54 and refactored in #53
